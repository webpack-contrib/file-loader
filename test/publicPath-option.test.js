import {
  compile,
  execute,
  getCompiler,
  normalizeErrors,
  readAsset,
} from './helpers';

describe('"publicPath" option', () => {
  it('should work without value', async () => {
    const compiler = getCompiler('simple.js');
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with "String" value', async () => {
    const compiler = getCompiler('simple.js', {
      publicPath: 'public_path/',
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with "String" value without trailing slash', async () => {
    const compiler = getCompiler('simple.js', {
      publicPath: 'public_path',
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with "String" value equal an URL', async () => {
    const compiler = getCompiler('simple.js', {
      publicPath: 'https://cdn.com/',
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with "Function" value and pass "url", "resourcePath" and "context" as arguments', async () => {
    expect.assertions(6);

    const compiler = getCompiler('simple.js', {
      publicPath(url, resourcePath, context) {
        expect(typeof url).toBe('string');
        expect(typeof resourcePath).toBe('string');
        expect(typeof context).toBe('string');

        return `public_path/${url}`;
      },
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });
});
