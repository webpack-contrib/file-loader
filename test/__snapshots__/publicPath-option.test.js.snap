// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"publicPath" option should work with "Function" value and pass "url", "resourcePath" and "context" as arguments: errors 1`] = `Array []`;

exports[`"publicPath" option should work with "Function" value and pass "url", "resourcePath" and "context" as arguments: result 1`] = `"public_path/39f5c21c1aee6ff21844c6e1d8251d97.png"`;

exports[`"publicPath" option should work with "Function" value and pass "url", "resourcePath" and "context" as arguments: warnings 1`] = `Array []`;

exports[`"publicPath" option should work with "String" value equal an URL: errors 1`] = `Array []`;

exports[`"publicPath" option should work with "String" value equal an URL: result 1`] = `"https://cdn.com/39f5c21c1aee6ff21844c6e1d8251d97.png"`;

exports[`"publicPath" option should work with "String" value equal an URL: warnings 1`] = `Array []`;

exports[`"publicPath" option should work with "String" value without trailing slash: errors 1`] = `Array []`;

exports[`"publicPath" option should work with "String" value without trailing slash: result 1`] = `"public_path/39f5c21c1aee6ff21844c6e1d8251d97.png"`;

exports[`"publicPath" option should work with "String" value without trailing slash: warnings 1`] = `Array []`;

exports[`"publicPath" option should work with "String" value: errors 1`] = `Array []`;

exports[`"publicPath" option should work with "String" value: result 1`] = `"public_path/39f5c21c1aee6ff21844c6e1d8251d97.png"`;

exports[`"publicPath" option should work with "String" value: warnings 1`] = `Array []`;

exports[`"publicPath" option should work without value: errors 1`] = `Array []`;

exports[`"publicPath" option should work without value: result 1`] = `"39f5c21c1aee6ff21844c6e1d8251d97.png"`;

exports[`"publicPath" option should work without value: warnings 1`] = `Array []`;
